### PR TITLE
Open / close main panel with swipes

### LINKED ISSUE #26: Add main panel gestures on mobile
Allow to swipe main panel up

### PR DESCRIPTION
**Pull request by baileys-li**
Fix #26

Minor changes for main panel logic

### CODE REVIEW

File: view/main/index.ts [hunk 1]
```diff
@@ -28,26 +25,6 @@ expand.addEventListener('click', () => {
...
-  emLogo.classList.add('is-hover')
-}
-const removeHoverLogo = (): void => {
-  emLogo.classList.remove('is-hover')
-}
-
-emText.addEventListener('mouseenter', addHoverLogo)
-emText.addEventListener('mouseleave', removeHoverLogo)
-emText.addEventListener('focus', addHoverLogo)
-emText.addEventListener('blur', removeHoverLogo)
```
Comment (baileys-li) [inline view/main/index.ts]: As I see, `:has` was already used. So it is make sense to synchronize the hover of these links with CSS only.
Besides, this is obviously not core functionality, so there is no point even think about polyfill that feature.

Comment (ai) [reply view/main/index.ts]: Good idea

File: package.json [hunk 2]
```diff
@@ -34,7 +34,7 @@
     "postcss-nesting": "^13.0.0",
     "postcss-opacity-percentage": "^2.0.0",
     "three": "~0.167.1",
-    "vite": "^5.3.5",
+    "vite": "^5.4.0",
```
Comment (ai) [inline package.json:37]: Can you revert this change? 

It is better to focus PR on a single task (also lock file changes could bring attacks).

File: view/main/logo.svg [hunk 3]
```diff
@@ -1 +1 @@
...
...l-5.29-5.34A10.6 10.6 0 0 0 20 2a2 2 0 1 0-4 0c0 3.81-2.19 6-6 6a2 2 0 1 0 0 4 10.6 10.6 0 0 0 5.66-1.51L21 15.78c-3.28 4-5 9.47-5 16.22v44zM64 40a10 10 0 1 1 0 20 10 10 0 0 1 0-20m-24 0a10 10 0 1 1 0 20 10 10 0 0 1 0-20"/><path d="M40 56a6 6 0 1 0 0-12 6 6 0 0 0 0 12m0-10a4 4 0 1 1 0 8 4 4 0 0 1 0-8m24 10a6 6 0 1 0 0-12 6 6 0 0 0 0 12m0-10a4 4 0 1 1 0 8 4 4 0 0 1 0-8m88 10a6 6 0 0 0 5.543-3.704 6.005 6.005 0 0 0-1.3-6.539 6.002 6.002 0 0 0-9.232.91A6.002 6.002 0 0 0 152 56m0-10a4 4 0 1 1 0 8 4 4 0 0 1 0-8m-24 10a6 6 0 0 0 5.543-3.704 6.005 6.005 0 0 0-1.3-6.539 6.002 6.002 0 0 0-9.232.91A6.002 6.002 0 0 0 128 56m0-10a4 4 0 1 1 0 8 4 4 0 0 1 0-8"/><path d="M176 76V63.8a10 10 0 0 0 0-19.6V32c0-15.25-8.75-24-24-24h-24c-15.25 0-24 8.75-24 24v12.2a10 10 0 0 0-5.753 16.128A10 10 0 0 0 104 63.8V76zm0-27.66a6 6 0 0 1 0 11.32zm-72 11.32a6 6 0 0 1 0-11.32zm24 .34a10.002 10.002 0 0 1-9.808-11.95 9.998 9.998 0 0 1 13.635-7.289A10 10 0 0 1 138 50a10 10 0 0 1-10 10m24 0a10.002 10.002 0 0 1-9.808-11.95 9.998 9.998 0 0 1 13.635-7.289A10 10 0 0 1 162 50a10 10 0 0 1-10 10m-42-30v-2.41l5.93 6 6.07-6 6 6 6-6 6 6 6-6 6 6 6-6 6 6 6-6v2.82l-5.93 6-6.07-6-6 6-6-6-6 6-6-6-6 6-6-6.05-6 5.9-6-6.21z"/></svg>
```
Comment (ai) [inline view/main/logo.svg:1]: If we want to change it, we need to change it for every SVG.

Let's think about it in separated PR.

Comment (baileys-li) [reply view/main/logo.svg:1]: I don't get it tbh 😅
`view/field/down.svg` use ` fill="currentcolor"` too. Or what do you mean?

Comment (ai) [reply view/main/logo.svg:1]: I mean that I do not see the changes in the file and the big PR with many changes is hard to review

Comment (baileys-li) [reply view/main/logo.svg:1]: Roger. Wait a little

File: view/main/index.pug [hunk 4]
```diff
@@ -1,5 +1,5 @@
-main.main
-  button.main_expand( aria-label="Expand or collapse panel" type="button" )
+main.main.layout_main#main-panel
+  button.main_expand( aria-label="Main panel" type="button" aria-expanded='true' aria-controls="main-panel" )
```
Comment (ai) [inline view/main/index.pug:2]: Why we change label? The new label doesn't describe the action

**Comment by ai** [comment]
I will be able to review it a little later. I need to see how it actually work.

**Comment by baileys-li** [comment]
Sure

File: view/main/index.pug (hunk 4 above)
Comment (baileys-li) [reply view/main/index.pug:2]: `aria-expanded` already say if element is collapsed or expanded. Screen reader even notify user on attribute change.

So with old label it sounds like oiled oil _(yeah, I didn't found English idiom)_

PS It's supposed to notify only that the [status has changed](https://doka.guide/a11y/aria-expanded/#:~:text=%D0%9A%D0%BE%D0%B3%D0%B4%D0%B0%20%D0%BF%D0%BE%D0%BB%D1%8C%D0%B7%D0%BE%D0%B2%D0%B0%D1%82%D0%B5%D0%BB%D1%8C%20%D1%80%D0%B0%D1%81%D0%BA%D1%80%D0%BE%D0%B5%D1%82%20%D1%81%D0%BF%D0%B8%D1%81%D0%BE%D0%BA%2C%20%D1%81%D0%BA%D1%80%D0%B8%D0%BD%D1%80%D0%B8%D0%B4%D0%B5%D1%80%20%D0%BF%D1%80%D0%BE%D1%81%D1%82%D0%BE%20%D1%81%D0%BE%D0%BE%D0%B1%D1%89%D0%B8%D1%82%20%C2%AB%D1%80%D0%B0%D0%B7%D0%B2%D1%91%D1%80%D0%BD%D1%83%D1%82%D0%BE%C2%BB%20%D0%B1%D0%B5%D0%B7%20%D0%BF%D0%BE%D0%B2%D1%82%D0%BE%D1%80%D0%B5%D0%BD%D0%B8%D1%8F%20%D0%B8%D0%BD%D1%84%D0%BE%D1%80%D0%BC%D0%B0%D1%86%D0%B8%D0%B8%20%D0%BE%20%D0%BA%D0%BD%D0%BE%D0%BF%D0%BA%D0%B5), but on my mac, Voice Over is re-reading the entire message:
> Main panel, expanded, button
> Main panel, collapsed, button

Comment (baileys-li) [reply view/main/index.pug:2]: I will ask in a11y chat how people deal with such cases

Comment (ai) [reply view/main/index.pug:2]: > Main panel, expanded, button
> Main panel, collapsed, button

Good note about `expanded`/`collapsed`. But I still prefer verb there.

Maybe `Toggle main panel`?

Comment (ai) [reply view/main/index.pug:2]: Also, let’s remove `main` since this word doesn’t say anything

**Comment by ai** [comment]
Looks great on real device. I am ready for merge when we will fix the label.

File: view/main/index.pug (hunk 4 above)
Comment (baileys-li) [reply view/main/index.pug:2]: Change to `toggle`.

PS Miss @glafirazhur confirmed that it's ok when a button in such contexts doesn't have a action label. For example, [accordion pattern](https://www.w3.org/WAI/ARIA/apg/patterns/accordion/examples/accordion/)

**Comment by ai** [comment]
Thanks. Amazing work.

**Comment by baileys-li** [comment]
Thanks
